src/Text/Pandoc/Writers/OpenDocument.hs

```diff
@@ -235,6 +243,12 @@ writeOpenDocument opts (Pandoc meta blocks) = do
                         meta
   ((body, metadata),s) <- flip runStateT
         defaultWriterState $ do
+           let collectInlineIdent (Image (ident,_,_) _ _) = [(ident,ImageRef)]
+               collectInlineIdent _                       = []
+           let collectBlockIdent (Header _ (ident,_,_) _)      = [(ident,HeaderRef)]
+               collectBlockIdent (Table (ident,_,_) _ _ _ _ _) = [(ident,TableRef)]
+               collectBlockIdent b                             = query collectInlineIdent b
+           modify $ \s -> s{ stIdentTypes = query collectBlockIdent blocks }
```

Comment:
Would this find an Image nested in a Table?
I forgot that Image is an inline and cannot be included into the `collectBlockIdent` function. I guess I was wrong, and we need at least two iterations.

Reply:
I've adjusted this now to make two iterations.